version: '3'

services:
  gnps-datasetcache-web:
    build: .
    ports:
      - "5234:5000"
    volumes:
      - ./database/:/app/database
      - ./temp/:/app/temp
      - ./logs/:/app/logs
    networks:
      - default
      - nginx-net
    restart: on-failure
    command: /app/run_server.sh
    environment:
      VIRTUAL_HOST: datasetcache.gnps2.org
      VIRTUAL_PORT: 5000
      LETSENCRYPT_HOST: datasetcache.gnps2.org
      LETSENCRYPT_EMAIL: mwang87@gmail.com
  
  # This does 
  gnps-datasetcache-beat:
    build: .
    volumes:
      - ./database/:/app/database
      - ./temp/:/app/temp
      - ./workflows/:/app/workflows
      - .env:/app/.env
    networks:
      - default
      - nginx-net
    restart: on-failure
    command: /app/run_worker_beat.sh

  gnps-datasetcache-worker:
    build: .
    volumes:
      - ./database/:/app/database
      - ./temp/:/app/temp
      - ./workflows/:/app/workflows
      - .env:/app/.env
    networks:
      - default
      - nginx-net
    restart: on-failure
    command: /app/run_worker_compute.sh
    
  gnps-datasetcache-redis:
    image: redis
    restart: on-failure
    networks:
      - default

  gnps-datasetcache-datasette:
    ports:
      - "5235:5234" # This port selection of the internal port matching the external port of the web server is to get around a datasette bug
    image: datasetteproject/datasette:0.64.6
    depends_on:
      - "gnps-datasetcache-web"
    volumes:
      - ./database/:/app/database:ro
    networks:
      - nginx-net
      - default
    restart: on-failure
    command: datasette -p 5234 -h 0.0.0.0 --setting base_url /datasette/ --setting max_returned_rows 50000 --setting sql_time_limit_ms 30000 --setting max_csv_mb 0 /app/database/database.db

networks:
  nginx-net:
    external:
      name: nginx-net